compose.yml

```diff
@@ -0,0 +1,105 @@
+# yredis demo stack - exposed services:
+#
+#   http://localhost:2222/    demos/auth-express app
+#   http://localhost:3333/    demos/blocksuite app
+#   http://localhost:3002/    yr-server (ydoc sync)
+#
+# these are just exposed for troubleshooting:
+#
+#   http://localhost:5432/    Postgres
+#   http://localhost:8001/    Redis analytics
+#   http://localhost:9001/    MinIO console
+
+x-vars:
+  shared: &shared
+    env_file: .env
+    # overrides to connect services inside of Docker overlay network
+    environment: &shared-env
+      REDIS: redis://redis:6379
+      POSTGRES: postgres://pg:5432
+      S3_ENDPOINT: s3
+      AUTH_PERM_CALLBACK: http://yr-authperm:5173/auth/perm
+      YDOC_UPDATE_CALLBACK: http://yr-docupdated:5173/ydoc
+
+services:
+  #
+  # demo apps
+  #
+  demo-codemirror: &demo-authexpress
```

Comment:
It would be nice if we could keep the main compose file small. IMHO it would be better to move the compose file to the demos and inherit from the main compose file, like I'm currently doing.